### PR TITLE
chore(maven): Plugin modernization

### PR DESCRIPTION
**Pull request by gounthar**
Hello `azure-sdk` developers!

This is an automated pull request created by the [Jenkins Plugin Modernizer](https://github.com/jenkinsci/plugin-modernizer-tool) tool. The tool has applied the following recipes to modernize the plugin:

<details>
    <summary>Add Jenkinsfile</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.AddOrModernizeJenkinsFile</em></p>
    <blockquote>Updates Jenkinsfile to build with recommended Java versions, platforms, and settings, or creates the file if absent. Also upgrades to a newer parent version, as the new version of the Jenkinsfille uses JDK17 and JDK21.</blockquote>
</details>

<details>
    <summary>Add plugins BOM</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.AddPluginsBom</em></p>
    <blockquote>Add the Jenkins BOM to the dependenciesManagement section of the pom.xml.
</blockquote>
</details>

<details>
    <summary>Upgrade parent version</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.UpgradeParentVersion</em></p>
    <blockquote>Upgrade the parent version to latest available</blockquote>
</details>

<details>
    <summary>Remove dependency version override</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.RemoveDependencyVersionOverride</em></p>
    <blockquote>Remove dependencies version override if managed from parent or bom</blockquote>
</details>

<details>
    <summary>Remove extra maven properties</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.RemoveExtraMavenProperties</em></p>
    <blockquote>Remove extra maven properties from the pom</blockquote>
</details>

<details>
    <summary>Migrate from Java 11 to Java 17</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.UpgradeToJava17</em></p>
    <blockquote>Migrate from Java 11 to Java 17</blockquote>
</details>

<details>
    <summary>Add Jenkinsfile</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.AddOrModernizeJenkinsFile</em></p>
    <blockquote>Updates Jenkinsfile to build with recommended Java versions, platforms, and settings, or creates the file if absent. Also upgrades to a newer parent version, as the new version of the Jenkinsfille uses JDK17 and JDK21.</blockquote>
</details>

<details>
    <summary>Upgrade to latest recommended core version and ensure the bom is matching the core version</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.UpgradeToRecommendCoreVersion</em></p>
    <blockquote>Upgrade to latest recommended core version and ensure the bom is matching the core version</blockquote>
</details>

<details>
    <summary>Convert JUnit 4 TestCase to JUnit Jupiter</summary>
    <p><em>io.jenkins.tools.pluginmodernizer.MigrateJUnitTestCase</em></p>
    <blockquote>Convert JUnit 4 TestCase to JUnit Jupiter</blockquote>
</details>

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
```
Comment (timja) [inline .github/dependabot.yml:1]: This repository uses renovate so the tool shouldn't try and add dependabot:

https://github.com/jenkinsci/azure-sdk-plugin/blob/master/.github/renovate.json

**Review by timja — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (gounthar) [reply .github/dependabot.yml:1]: Sure thing, thank you. 👍

Comment (timja) [reply .github/dependabot.yml]: thanks FWIW more general feedback for the tool, especially if we will be running it automatically in the future

**Review by timja — APPROVED** [review]

**Comment by gounthar** [comment]
Thanks!

**Pull request merged**